examples/v2/project_creation/project.py

```diff
@@ -98,3 +101,35 @@ def GenerateConfig(context):
     })
 
   return {'resources': resources}
+
+def MergeCallingServiceAccountWithOwnerPermissinsIntoBindings(env, properties):
+  """ A helper function that merges the acting service account of the project
+      creator as an owner of the project being created
+  """
+  service_account = ('serviceAccount:{0}@cloudservices.gserviceaccount.com'
+                     .format(env['project_number']))
+  set_creator_sa_as_owner = {
+      'role': 'roles/owner',
+      'members': [
+          service_account,
+      ]
+  }
+  if ('iam-policy' not in properties or
+      'bindings' not in properties['iam-policy']):
+    return {
+        'bindings': [
+            set_creator_sa_as_owner,
+        ]
+    }
+  bindings = copy.deepcopy(properties['iam-policy']['bindings'])
+  merged = False
+  for binding in bindings:
```

Comment:
I _think_ (99% certain) if you structured your set_creator_sa_as_owner dictionary just like the bindings one, then you could replace the merge logic with

`
set_creator_sa_as_owner.update(bindings)
`

and then everything in bindings will overwrite the contents of set_creator_sa_as_owner where appropriate and add everything else.